client/src/controllers/BlockController.ts

```diff
@@ -0,0 +1,76 @@
+import { Controller } from '@hotwired/stimulus';
+
+declare global {
+  interface Window {
+    initBlockWidget?: (id: string) => void; // Declare the initBlockWidget function as optional on the Window interface
+    telepath: any;
+  }
+}
+
+/**
+ * Adds the ability to unpack a Telepath object and render it on the controlled element.
+ *
+ * @example
+ * <div
+ *  id="some-id"
+ *  data-controller="w-block"
+ *  data-w-block-data-value='{"some":"json"}'
+ * >
+ * </div>
+ */
+export class BlockController extends Controller<HTMLElement> {
+  static values = {
+    initial: { type: Array, default: [] },
+    error: { type: Object, default: {} },
+    data: { type: Object, default: {} },
+  };
+
+  /** Initial value. */
+  declare initialValue: Array<string>;
+  /** Optional error value. */
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  declare errorValue: Object;
+  /** Block data value to unpack with Telepath. */
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  declare dataValue: Object;
```

Comment:
The Typescript type should be `object` (lowercase). 

It's only the Stimulus value type that should be the upper case `Object`. 

The reason for this is that the Stimulus one needs a constructor class, but Typescript does not use that.

Reply:
Oh, okie. I got it.